lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -57,19 +52,29 @@ const SessionAuth: React.FC<PropsWithChildren<Props>> = ({ children, ...props })
     const parentSessionContext = useContext(SessionContext);
 
     // assign the parent context here itself so that there is no flicker in the UI
-    const [context, setContext] = useState<SessionContextType | undefined>(
-        hasParentProvider(parentSessionContext) ? parentSessionContext : undefined
+    const [context, setContext] = useState<SessionContextType>(
+        !isDefaultContext(parentSessionContext) ? parentSessionContext : { loading: true }
     );
 
-    const session = useRef(Session.getInstanceOrThrow());
+    const session = useRef<Session>();
     const userContext = useUserContext();
 
+    const redirectToLogin = useCallback(() => {
+        if (props.redirectToLogin !== undefined) {
+            props.redirectToLogin();
+        } else {
+            void session.current!.redirectToAuthWithRedirectToPath(
+                SuperTokens.getInstanceOrThrow().getReactRouterDomWithCustomHistory()?.useHistoryCustom()
+            );
```

Comment:
We should change this to call a top level getRedirectionURL function (on supertokens.init level) with a context like:
```
{
   action: "TO_AUTH",
   showSignIn: boolean | undefined
}
```

And exposed `SuperTokens.redirectToAuth({...})` takes the same stuff as what we have right now for individual recipes. It should also set the right query params from the return of `getRedirectionURL` (if the user has not set them already)

We should remove the recipe level redirectToAuth functions, and the SIGN_IN_UP action associated with them.